crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -76,6 +76,18 @@ pub(crate) struct ObjectKey {
     version: u64,
 }
 
+/// The basic unit of storage in Sui is object. In contrast to many other blockchains,
+/// where storage is centered around address and each address contains a key-value store,
+/// Sui's storage is centered around objects. Sui objects have one of the following primary states:
+///
+///    Immutable - the object cannot be modified.
+///    Mutable - the object can be changed.
+///
+/// Further, mutable objects are divided into these categories:
+///
+///    Owned - the object can be modified only by its owner.
+///    Shared - the object can be modified by anyone.
+/// Immutable objects do not need this distinction because they have no owner.
```

Comment:
I think this is more information than necessary and repeats some information from the Object's fields.
I feel like we can just get away with something like, (very rough) "An object in Sui is a package (set of Move bytecode modules) or object (typed data structure with fields) with additional metadata detailing its id, version, transaction digest, owner field indicating how this object can be accessed." something like that

Reply:
I tend to agree with @wlmyng that this documentation is probably better aligned with the object owner/kind field, and not `Object` itself. I would also avoid explaining a concept with reference to some other concept that it's not (the account model), in reference material.

Also note that owned objects can also only be read (on-chain) by their owner, (i.e. the restriction on use is not limited to modification).

Reply:
Thanks for the valuable feedback here!